tests/bot/cogs/test_snekbox.py

```diff
@@ -0,0 +1,363 @@
+
+    @async_test
+    async def test_eval_command_evaluate_twice(self):
+        """Test the eval command re-evaluation procedure."""
+        ctx = MockContext()
+        ctx.message = MockMessage()
+        ctx.message.content = '!e MyAwesomeCode'
+        updated_msg = MockMessage()
+        updated_msg .content = '!e MyAwesomeCode-2'
+        response_msg = MockMessage()
+        response_msg.delete = AsyncMock()
+        ctx.send = AsyncMock(return_value=response_msg)
+        ctx.author.mention = '@LemonLemonishBeard#0042'
+        ctx.typing = MagicMock(return_value=AsyncContextManagerMock(None))
+        self.cog.post_eval = AsyncMock(return_value={'stdout': '', 'returncode': 0})
+        self.cog.get_results_message = MagicMock(return_value=('Return code 0', ''))
+        self.cog.get_status_emoji = MagicMock(return_value=':yay!:')
+        self.cog.format_output = AsyncMock(return_value=('[No output]', None))
+        self.bot.wait_for.side_effect = ((None, updated_msg), None, asyncio.TimeoutError)
+
+        await self.cog.eval_command.callback(self.cog, ctx=ctx, code='MyAwesomeCode')
+
+        self.cog.post_eval.assert_has_calls((call('MyAwesomeCode'), call('MyAwesomeCode-2')))
+
+        # Multiplied by 2 because we expect it to be called twice
+        ctx.send.assert_has_calls(
+            [call('@LemonLemonishBeard#0042 :yay!: Return code 0.\n\n```py\n[No output]\n```')] * 2
+        )
+        self.cog.get_status_emoji.assert_has_calls([call({'stdout': '', 'returncode': 0})] * 2)
+        self.cog.get_results_message.assert_has_calls([call({'stdout': '', 'returncode': 0})] * 2)
+        self.cog.format_output.assert_has_calls([call('')] * 2)
+
+        self.bot.wait_for.has_calls(
+            call('message_edit', check=snekbox.predicate_eval_message_edit, timeout=10),
+            call('reaction_add', check=snekbox.predicate_eval_emoji_reaction, timeout=10)
+        )
+        ctx.message.add_reaction.assert_called_once_with('🔁')
+        ctx.message.clear_reactions.assert_called()
+        response_msg.delete.assert_called_once()
+
+    def test_predicate_eval_message_edit(self):
+        """Test the predicate_eval_message_edit function."""
+        msg0 = MockMessage()
+        msg0.id = 1
+        msg0.content = 'abc'
+        msg1 = MockMessage()
+        msg1.id = 2
+        msg1.content = 'abcdef'
+        msg2 = MockMessage()
+        msg2.id = 1
+        msg2.content = 'abcdef'
+
+        cases = (
+            (msg0, msg0, False, 'same ID, same content'),
+            (msg0, msg1, False, 'different ID, different content'),
+            (msg0, msg2, True, 'same ID, different content')
+        )
+        for ctx_msg, new_msg, expected, testname in cases:
+            with self.subTest(msg=f'Messages with {testname} return {expected}'):
+                ctx = MockContext()
+                ctx.message = ctx_msg
+                self.assertEqual(snekbox.predicate_eval_message_edit(ctx, ctx_msg, new_msg), expected)
```

Comment:
Reads better as separate lines.
```suggestion
                actual = snekbox.predicate_eval_message_edit(ctx, ctx_msg, new_msg)
                self.assertEqual(actual, expected)
```